client/src/client/user_interface/__init__.py

```diff
@@ -201,10 +209,14 @@ def validate_password(self, password: str) -> bool:
         password_without_whitespace = "".join(password.split())
         return password == password_without_whitespace and len(password) >= 6
 
-    def validate_mnemonic(self, password: str) -> bool:
-        """Validate if given one time password is valid."""
-        password_without_whitespace = "".join(password.split())
-        return password == password_without_whitespace and len(password) == 10
+    def validate_mnemonic(self, mnemonic: str) -> bool:
+        """Validate if given mnemonic is valid."""
+        mnemonic_without_whitespace = "".join(mnemonic.split())
+        return attempt_decrypting_priv_key_backup_files(
```

Comment:
I don't think I like that... this should be just "client-side check", the validity of mnemonic should be checked in the higher level (client)

Reply:
In my opinion, it should look like this:
`client.__init__()`:
```
if user_passphrase == "~":
    succesfull = False
    while not succesfull:
            mnemonic = self.ui.blocking_prompt(
                PasswordRecoveryState.PROMPT_MNEMONIC
            )
            if attempt_decrypting_priv_key_backup_files(mnemonic):
                new_passphrase = self.ui.blocking_prompt(
                    PasswordRecoveryState.PROMPT_NEW_PASSWORD
                )
                encrypt_password(new_passphrase)
                succesfull = True
            else:
                continue
    assert mnemonic and new_passphrase is not None
                    ```
UI shouldn't do anything, it's only there to provide input

Reply:
On second thought: This shouldn't even be a while loop, just one chance, and if you're wrong you just go back to password prompt maybe